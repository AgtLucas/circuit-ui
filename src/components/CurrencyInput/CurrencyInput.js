import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, cx } from 'react-emotion';
import { withTheme } from 'emotion-theming';

import IconInput from '../IconInput';
import { themePropType } from '../../util/shared-prop-types';

const currencyBaseStyles = ({ theme }) => css`
  color: ${theme.colors.b700};
`;

const CurrencyIcon = styled('span')`
  ${currencyBaseStyles};
`;

/**
 * SearchInput component for forms.
 */
const CurrencyInput = ({ currencyPosition, currency, theme, ...props }) => (
  <IconInput
    iconLeft={({ className }) =>
      currencyPosition === 'left' && (
        <CurrencyIcon className={className}>{currency}</CurrencyIcon>
      )
    }
    iconRight={({ className }) =>
      currencyPosition === 'right' && (
        <CurrencyIcon className={className}>{currency}</CurrencyIcon>
      )
    }
    {...{ theme, ...props }}
  />
);

CurrencyInput.propTypes = {
  theme: themePropType.isRequired,
  currencyPosition: PropTypes.oneOf(['left', 'right']),
  textAlign: PropTypes.oneOf(['left', 'right']),
  currency: PropTypes.string
};

CurrencyInput.defaultProps = {
  currencyPosition: 'right',
  textAlign: 'right',
  currency: 'â‚¬'
};

/**
 * @component
 */
export default withTheme(CurrencyInput);
