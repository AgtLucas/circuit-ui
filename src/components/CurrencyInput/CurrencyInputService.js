import { curry } from 'lodash/fp';

import { formatAmountForLocale } from '../../util/currency';

export const parseAmount = value => {
  if (!value || !value.length) {
    return value;
  }

  const digits = value.replace(/[^\d]/g, '');

  if (digits === '') {
    return '0.00';
  }

  return `${parseInt(digits, 10) / 100}`;
};

export const formatAmount = curry(
  (currency, locale, amount) =>
    amount && amount.length
      ? formatAmountForLocale(amount, currency, locale)
      : ''
);
