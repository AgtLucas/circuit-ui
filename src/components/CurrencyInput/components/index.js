export { default as SimpleCurrencyInput } from './SimpleCurrencyInput';
