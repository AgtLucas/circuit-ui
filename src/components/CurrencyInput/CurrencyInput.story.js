import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import withTests from '../../util/withTests';
import CurrencyInput from '.';

storiesOf('CurrencyInput', module)
  .addDecorator(withTests('CurrencyInput'))
  .add(
    'Default CurrencyInput',
    withInfo()(() => <CurrencyInput placeholder="123.45" />)
  )
  .add(
    'Invalid CurrencyInput',
    withInfo()(() => <CurrencyInput placeholder="123.45" invalid />)
  )
  .add(
    'Optional CurrencyInput',
    withInfo()(() => <CurrencyInput placeholder="123.45" optional />)
  )
  .add(
    'Disabled CurrencyInput',
    withInfo()(() => <CurrencyInput placeholder="123.45" disabled />)
  )
  .add(
    'USD CurrencyInput',
    withInfo()(() => (
      <CurrencyInput
        currency="$"
        placeholder="1,234.45"
        currencyPosition="left"
        textAlign="left"
      />
    ))
  )
  .add(
    'BGN CurrencyInput',
    withInfo()(() => <CurrencyInput currency="лв." placeholder="1 234,56" />)
  )
  .add(
    'CLP CurrencyInput',
    withInfo()(() => (
      <CurrencyInput
        currency="$"
        placeholder="1.234"
        currencyPosition="left"
        textAlign="left"
      />
    ))
  );
