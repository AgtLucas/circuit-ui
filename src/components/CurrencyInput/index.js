import CurrencyInput from './CurrencyInput';

export * from './components';
export * from './CurrencyInputService';
export default CurrencyInput;
