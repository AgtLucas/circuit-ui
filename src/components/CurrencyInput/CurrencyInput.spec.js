import React from 'react';

import CurrencyInput from '.';

describe('CurrencyInput', () => {
  it.skip('should', () => {});
});
