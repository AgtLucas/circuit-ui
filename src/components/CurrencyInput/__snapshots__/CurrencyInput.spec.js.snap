// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput should render with BGN currency styles 1`] = `
.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
}

.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  text-align: right;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 12px;
  left: auto;
}

<div
  className="circuit-4 circuit-5"
  disabled={false}
>
  <input
    className="circuit-0 circuit-1"
    disabled={false}
    placeholder="123,45"
  />
  <span
    className="circuit-2 circuit-3"
  >
    лв.
  </span>
</div>
`;

exports[`CurrencyInput should render with USD currency styles 1`] = `
.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
}

.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  left: 12px;
  right: auto;
}

<div
  className="circuit-4 circuit-5"
  disabled={false}
>
  <input
    className="circuit-0 circuit-1"
    disabled={false}
    placeholder="123.45"
  />
  <span
    className="circuit-2 circuit-3"
  >
    $
  </span>
</div>
`;

exports[`CurrencyInput should render with default styles 1`] = `
.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
}

.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  text-align: right;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 12px;
  left: auto;
}

<div
  className="circuit-4 circuit-5"
  disabled={false}
>
  <input
    className="circuit-0 circuit-1"
    disabled={false}
  />
  <span
    className="circuit-2 circuit-3"
  >
    €
  </span>
</div>
`;

exports[`CurrencyInput should render with disabled styles when passed the disabled prop 1`] = `
.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  text-align: right;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 12px;
  left: auto;
}

.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
  opacity: 0.5;
  pointer-events: none;
  box-shadow: none;
}

<div
  className="circuit-4 circuit-5"
  disabled={true}
>
  <input
    className="circuit-0 circuit-1"
    disabled={true}
  />
  <span
    className="circuit-2 circuit-3"
  >
    €
  </span>
</div>
`;

exports[`CurrencyInput should render with invalid styles when passed the invalid prop 1`] = `
.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 12px;
  left: auto;
}

.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  text-align: right;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

.circuit-0:not(:focus) {
  border-color: #FFAF9F;
}

.circuit-0:not(:focus)::-webkit-input-placeholder {
  color: #FFAF9F;
}

.circuit-0:not(:focus)::-moz-placeholder {
  color: #FFAF9F;
}

.circuit-0:not(:focus):-ms-input-placeholder {
  color: #FFAF9F;
}

.circuit-0:not(:focus)::placeholder {
  color: #FFAF9F;
}

<div
  className="circuit-4 circuit-5"
  disabled={false}
>
  <input
    className="circuit-0 circuit-1"
    disabled={false}
  />
  <span
    className="circuit-2 circuit-3"
  >
    €
  </span>
</div>
`;

exports[`CurrencyInput should render with optional styles when passed the optional prop 1`] = `
.circuit-4 {
  color: #212933;
  display: block;
  position: relative;
  margin-bottom: 16px;
}

.circuit-2 {
  color: #2567D8;
  line-height: 16px;
  position: absolute;
  height: 16px;
  width: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 12px;
  left: auto;
}

.circuit-0 {
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #D8DDE1;
  border-radius: 4px;
  box-shadow: inset 0 1px 2px 0 rgba(102,113,123,0.12);
  padding: 8px 12px;
  width: 100%;
  font-size: 15px;
  line-height: 24px;
  background-color: #FAFBFC;
  border-style: dashed;
  box-shadow: none;
  text-align: right;
  padding-left: calc( 12px + 16px + 12px );
  padding-right: calc( 12px + 16px + 12px );
}

.circuit-0:focus,
.circuit-0:active {
  border: 1px solid #3388FF;
  outline: none;
}

.circuit-0::-webkit-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::-moz-placeholder {
  color: #9DA7B1;
}

.circuit-0:-ms-input-placeholder {
  color: #9DA7B1;
}

.circuit-0::placeholder {
  color: #9DA7B1;
}

<div
  className="circuit-4 circuit-5"
  disabled={false}
>
  <input
    className="circuit-0 circuit-1"
    disabled={false}
  />
  <span
    className="circuit-2 circuit-3"
  >
    €
  </span>
</div>
`;
